### PR TITLE
chore: bump swc and webpack-dev-server for examples

### LINKED ISSUE #496: SWC hot reload doesn't work when arrow function is used
Whenever I use `arrow function` instead of `function`, hot reload doesn't work. See screenshot and how to repro below: 

image

Repro:
1. clone react-refresh-webpack-plugin
2. navigate to examples/typescript-with-swc
3. Change 
```
function App() {
  return (
    <div>
      <ClassDefault />
      <ClassNamed />
      <FunctionDefault />
      <FunctionNamed />
      <Suspense fallback={<h1>Loading</h1>}>
        <LazyComponent />
      </Suspense>
    </div>
  );
}
```
to
```
const App= () => {
  return (
    <div>
      {/* added */}
     <div> test </div> 
      <ClassDefault />
      <ClassNamed />
      <FunctionDefault />
      <FunctionNamed />
      <Suspense fallback={<h1>Loading</h1>}>
        <LazyComponent />
      </Suspense>
    </div>
  );
}
```
4. Change `test` text to different text

### PR DESCRIPTION
**Pull request by Austaras**
fixes #496

### CODE REVIEW

File: examples/typescript-with-swc/package.json [hunk 1]
```diff
@@ -8,9 +8,10 @@
   },
   "devDependencies": {
     "@pmmmwh/react-refresh-webpack-plugin": "^0.5.0",
-    "@swc/core": "^1.2.85",
+    "@swc/core": "^1.2.88",
     "@types/react": "^17.0.20",
     "@types/react-dom": "^17.0.9",
+    "core-js": "^3.17.3",
```
Comment (pmmmwh) [inline examples/typescript-with-swc/package.json:14]: Is this intentional?

File: examples/typescript-with-swc/src/App.tsx [hunk 2]
```diff
@@ -13,6 +14,7 @@ function App() {
       <ClassNamed />
       <FunctionDefault />
       <FunctionNamed />
+      <ArrowFunction />
```
Comment (pmmmwh) [inline examples/typescript-with-swc/src/App.tsx:17]: Is it possible to add this for all examples 🙏🏻 ?

File: examples/typescript-with-swc/package.json (hunk 1 above)
Comment (Austaras) [reply examples/typescript-with-swc/package.json:14]: Yes. `swc` need explicit `core-js` dependency for compat.

File: examples/typescript-with-swc/src/App.tsx (hunk 2 above)
Comment (Austaras) [reply examples/typescript-with-swc/src/App.tsx:17]: Sure, will do

**Review by pmmmwh — APPROVED** [review]

**Comment by pmmmwh** [comment]
Thanks for the PR!

**Pull request merged**